CONTRIBUTING.md

```diff
@@ -1,10 +1,63 @@
+- If the [Github issue](https://github.com/jamulussoftware/jamulus/issues) for your feature/bug fix already exists, write a message in that issue saying you want to work on it. 
+- Otherwise, please [post on the GitHub Discussions](https://github.com/jamulussoftware/jamulus/discussions) and say that you are planning to do some coding and explain why. Then we can discuss the specification. 
+- Please only start coding after we have agreed to a specification to avoid putting lots of work into something which may not get accepted later.
+
+## Jamulus project/source code general principles
+
+### Stability
+This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+
+### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus. 
+
+### Source code consistency
+- Respect the existing code style: Tab size=4, insert spaces.
+- Space before and after `(` and `)`, except no space between `)` and `;`, and no space before an empty `()`.
+- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, on separate lines.
+
+Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+
+Please check your Github fork. It should build your branch for all the supported platforms (though not all supported platform versions) when you push. Make sure there are no errors. If there are, resolve them before creating a pull request.
+
+If there are conflicts with jamulussoftware/jamulus showing when you go to raise the pull request, resolve those locally:
+~~~
+git checkout master ;# checkout your local master
+git pull upstream master --tags ;# refresh from jamulussoftware/jamulus (i.e. use your upstream remote name)
+git push --tags --force ;# update your fork's master
+git checkout - ;# back to your working branch
+git rebase master ;# replay your changes onto the latest code
+git push --force ;# update your working branch on your fork
+~~~
+
+### Supported platforms
+We support the following platforms and versions:
+- Windows 10 <!-- Is this the only version? Do we have to specify a specific sub-version? -->
+- macOS <!-- Which versions? -->
+- Ubuntu Linux 18.04 or later, Debian 10 or later, most Linux flavors with recent enough Qt versions
+<!-- Do we support BSD? I think a recent discussion was about FreeBSD? -->
+<!-- Should we already list Android? If so, what platforms/versions? -->
+
+The minimum Qt version to support is 5.X <-- What is it? -->
```

Comment:
```suggestion
The minimum Qt version to support is 5.X <!-- What is it? -->
```

Still don't know what to fill in, but this should be documented, IMO. :(

Reply:
We support 5.12.2 on all platforms except MacOS, where we need to support 5.9 -- so that means Qt 5.9 as a latest version for build.

It should be noted that _code style_ in a file may be Qt 4.x.  Normally, "stick to existing style" applies.  However, if making large-scale changes, then updating to Qt 5.9 is encouraged.